client/src/pages/locations/GeoLocation.js

```diff
@@ -148,27 +138,30 @@ const MGRSFormField = ({
           <Field
             name="displayedCoordinate"
             component={FieldHelper.InputFieldNoLabel}
-            value={mgrs}
-            onChange={e => setMgrs(e.target.value)}
             onBlur={e => {
-              const newLatLng = convertMGRSToLatLng(mgrs)
-              setValues({
-                displayedCoordinate: e.target.value,
-                lat: newLatLng[0],
-                lng: newLatLng[1]
-              })
               setFieldTouched("displayedCoordinate", true, false)
+              const newLatLng = convertMGRSToLatLng(e.target.value)
+              setFieldValue(
```

Comment:
Yes, normally it would. But i am updating lat,lng fields with MGRS's onChange, which overwrites the Formik's onChange so we have to update manually. It is unnecessary in the onBlur though, yes.